### ISSUE TITLE
[java] SimplifyBooleanReturns also reports the last if/else in a longer if/else-chain

### ISSUE THREAD

**Issue by kris-scheibe** [pmd/pmd#5219]
**Affects PMD Version: 7.5** 

**Rule: SimplifyBooleanReturns**

Please provide the rule name and a link to the rule documentation:
SimplifyBooleanReturns
https://docs.pmd-code.org/latest/pmd_rules_java_design.html#simplifybooleanreturns

**Description:**

SimplifyBooleanReturns reports a violation for the last "if" in the longer if/else-block in the code sample below.

**Code Sample demonstrating the issue:**

```java
private static final Set<String> TRUE_STRINGS = Set.of("true", "on", "yes", "1");
private static final Set<String> FALSE_STRINGS = Set.of("false", "off", "no", "0");

boolean toBool(String stringToConvert, boolean defaultValue) {
    if (stringToConvert.isEmpty()) {
        return defaultValue;
    } else if (TRUE_STRINGS.contains(stringToConvert)) {
        return true;
    } else if (FALSE_STRINGS.contains(stringToConvert)) { // PMD: This if statement can be replaced by `return !{condition} || {elseBranch};`
        return false;
    } else {
        return defaultValue;
    }
}
```

**Expected outcome:**

IMO, this rule should only trigger, if the whole if/else-block can be replaced with a single expression. In this case, the first two returns would remain unchanged, and only the the last two returns get folded into a single return-expression.
I think this is a bad recommendation and would make the whole if/else-block harder to read. 
The documentation also only shows replacing the entire if/else-statement, but this might be for brevity.

So, replacing a statement like 

```java
if (TRUE_STRINGS.contains(stringToConvert)) {
    return true;
} else {
    return defaultValue;
}
```
with
```java
return TRUE_STRINGS.contains(stringToConvert) || defaultValue;
```
is a simplification, recommending the change above is not.

**Running PMD through:** *Maven*
